java/concept/answer/QueryAnswerImpl.java

```diff
@@ -19,24 +19,31 @@
 
 package com.typedb.driver.concept.answer;
 
+import com.typedb.driver.api.QueryType;
 import com.typedb.driver.api.answer.QueryAnswer;
-import com.typedb.driver.common.NativeObject;
 import com.typedb.driver.common.exception.TypeDBDriverException;
 
 import static com.typedb.driver.common.exception.ErrorMessage.Internal.UNEXPECTED_NATIVE_VALUE;
-import static com.typedb.driver.jni.typedb_driver.query_answer_is_concept_rows_stream;
-import static com.typedb.driver.jni.typedb_driver.query_answer_is_concept_trees_stream;
+import static com.typedb.driver.jni.typedb_driver.query_answer_get_query_type;
+import static com.typedb.driver.jni.typedb_driver.query_answer_is_concept_document_stream;
+import static com.typedb.driver.jni.typedb_driver.query_answer_is_concept_row_stream;
 import static com.typedb.driver.jni.typedb_driver.query_answer_is_ok;
 
-public abstract class QueryAnswerImpl extends NativeObject<com.typedb.driver.jni.QueryAnswer> implements QueryAnswer {
+public abstract class QueryAnswerImpl implements QueryAnswer {
+    public final QueryType queryType;
+
     protected QueryAnswerImpl(com.typedb.driver.jni.QueryAnswer answer) {
-        super(answer);
+        queryType = QueryType.of(query_answer_get_query_type(answer));
```

Comment:
There was a bug when I initially implemented `QueryAnswer` as `NativeObject` and then proceeded to call `intoRows` inside `ConceptRowIteratorImpl`, which freed the `QueryAnswer`. 
Now, we don't store the native `QueryAnswer` anywhere on the Java side (same for Python), it's automatically unwrapped to something else, and that's it.

To access `QueryType` from non-consumed streams, I copy the small `enum` on the init, and we can continue doing it for any small metadata we want to expose. However, we might need to stop unwrapping the `QueryAnswer` to a specific ok/row/document on the init if we want to be able to call for these methods if more data appear (which would make the work with the native object even more complicated, so we leave it like this).